.github/workflows/docker-image.yml

```diff
@@ -24,12 +25,18 @@ jobs:
         username: ${{ secrets.DOCKER_USERNAME }}
         password: ${{ secrets.DOCKER_PASSWORD }}
 
-    - name: Build and tag the Docker image
-      if: github.ref != 'refs/heads/master'
-      run: docker build . --file Dockerfile --tag nuonic/node-python-awscli:${{ steps.branch-name.outputs.branch }}
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
 
-    - name: Build, tag and push the image
-      if: github.ref == 'refs/heads/master'
-      run: |
-          make build tag push
+    - name: Build Docker image
+      uses: docker/build-push-action@v5
+      with:
+        context: .
+        load: true
+        tags: nuonic/node-python-awscli:latest
```

Comment:
https://github.com/docker/build-push-action
This step will build AND push AND tag as latest the docker image, unconditionally, on every branch.

That makes the next step "make tag push" unnecessary.

I suggest you tag with the branch name here instead of latest, and conditionally tag as latest on the next step.